tools/lz4/Makefile

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (C) 2022 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=lz4
+PKG_VERSION:=1.9.4
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/lz4/lz4/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=0b0e3aa07c8c063ddf40b082bdf7e37a1562bda40a0ff5272957f3e987e0e54b
```

Comment:
please do not split liblz4 and lz4 into separate tools packages. 
The split will increase the maintenance effort too much and currently both parts are always needed at the same time anyway.

Reply:
squashfskit4 only uses the `liblz4` and `liblzo` when selected.  `lz4` or `lzop` are only used for if you pick LZ4 or LZO initramfs and never any other time.  The system versions of everything are used everywhere but image building.

I will put myself on as maintainer it's not hard.

Reply:
What would be better is if a host tools Makefile could have both halves in one file like a target package can.  But vimdiff almost makes it like editing one file, the diffs aren't huge and now that I came up with how to call the package something other than what the tarfile is called it works pretty slick.

Oh and there are some other existing lib separate from exec which was the precedent I was following.  If these need to be merged then so does anything else that is split (consistency uber alles).

Reply:
Okay now that I am more fully awake, I don't care either way, and it is a lot easier to just build a bunch of junk we don't actually need especially on the host side where there isn't a space or speed problem.  I'll merge the lib/exec splits.  Really just half a dozen vs six.  And I did a lot of work to make it happen because it seemed like the right way based on pattern matching and the dependency map.  But hey sometimes work becomes exercise instead, no loss, I got some experience along the way and found unrelated nitpicks too.

Is it preferable to force the upstream source do what makes sense via patches like I have, or keep making custom install phase macros?  I have been trying to only use the Default handlers whenever possible and patch the source to do the right things in its own `install` target while accepting `PREFIX` like the ad-hoc standard would expect.  Then there seems less duplication and less to get wrong, but you do have to go dig into the build_dir to see what the Makefile does rather than it just being right there, so it seems like it might be another conflict of perspective.  Oh and I bet I'm numbering patches wrong too, be sure to point that out if so.  I could only find directions about kernel patches and what the numbering means, is it roughly similar?  Some packages have four-digit numbering...

Reply:
`lz4` is now unified lib+bin package because they share the same source package, `liblzo` remains separate from `lzop` because they are actually separate projects.